metrics/metrics-api/src/main/java/org/apache/james/metrics/api/NoopMetricFactory.java

```diff
@@ -55,4 +57,9 @@ public long stopAndPublish() {
             return 0;
         }
     }
+
+    @Override
+    public <T> T withMetric(String name, Supplier<T> operation) {
+        return operation.get();
```

Comment:
is it still a Noop if it runs the operation?

Reply:
imo it stays "transparent" to the operation then it makes sens to keep the NOOP naming.